Neos.Flow/Tests/Unit/ResourceManagement/Streams/PackageStreamWrapperTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+namespace Neos\Flow\Tests\Unit\ResourceManagement\Streams;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Package\FlowPackageInterface;
+use Neos\Flow\ResourceManagement\Exception;
+use Neos\Flow\ResourceManagement\Streams\PackageStreamWrapper;
+use org\bovigo\vfs\vfsStream;
+use Neos\Flow\Package\PackageManager;
+use Neos\Flow\Tests\UnitTestCase;
+
+/**
+ * Tests for the PackageStreamWrapper class
+ */
+class PackageStreamWrapperTest extends UnitTestCase
+{
+    /**
+     * @var PackageStreamWrapper
+     */
+    protected $packageStreamWrapper;
+
+    /**
+     * @var PackageManager|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $mockPackageManager;
+
+    protected function setUp(): void
+    {
+        vfsStream::setup('Foo');
+
+        $this->packageStreamWrapper = new PackageStreamWrapper();
+
+        $this->mockPackageManager = $this->createMock(PackageManager::class);
+        $this->inject($this->packageStreamWrapper, 'packageManager', $this->mockPackageManager);
+    }
+
+    /**
+     * @test
+     */
+    public function openThrowsExceptionForInvalidScheme()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $openedPathAndFilename = '';
+        $this->packageStreamWrapper->open('invalid-scheme://foo/bar', 'r', 0, $openedPathAndFilename);
+    }
+
+    public function providePathesWithForbiddenUpwardsTraversel(): array
+    {
+        return [
+            ['package://Some.Package/../bar', true],
+            ['package://Some.Package/foo/../bar', true],
+            ['package://Some.Package/bar/..', false],
+            ['package://Some.Package/..', false],
```

Comment:
Shouldn't that be an exceptional case?

Reply:
No because this happens while iterating over a folder. Since one cannot only traverse up but not down again to another file/folder this should be safe

Reply:
What about `package://Foo.Bar/../../../`?

Reply:
Is forbidden because of upwards traversal in the middle of the path ... added example to test to clarify that

Reply:
But `['package://Some.Package/..', false]` still looks wrong to me. It traverses outside the package!